src/SprykerSdk/Acp/Schema/SchemaExtender.php

```diff
@@ -0,0 +1,155 @@
+
+        if ($configurationSchemaGeneratedMessage) {
+            $createDefaultEndpointsResponseTransfer->addError($configurationSchemaGeneratedMessage);
+        }
+
+        return $createDefaultEndpointsResponseTransfer;
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return \Transfer\CreateDefaultEndpointsResponseTransfer
+     */
+    protected function addDefaultEndpointsToSchema(
+        CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+    ): CreateDefaultEndpointsResponseTransfer {
+        $result = new CreateDefaultEndpointsResponseTransfer();
+
+        $schema = $this->getDefaultEndpointsSchema($createDefaultEndpointsRequestTransfer);
+
+        $process = new Process([
+            'vendor/bin/syncapi',
+            'schema:openapi:update',
+            '-f',
+            $createDefaultEndpointsRequestTransfer->getSchemaFile(),
+            $this->schemaWriter->writeSchemaToJsonString($schema),
+        ]);
+        $process->run();
+
+        if (!$process->isSuccessful()) {
+            $result->addError($this->createMessage(sprintf('Updating openapi schema failed, error output: %s', $process->getErrorOutput())));
+
+            return $result;
+        }
+
+        if ($createDefaultEndpointsRequestTransfer->getAddLocal()) {
+            $registryFileCopied = $this->copyRegistryLocalFile($createDefaultEndpointsRequestTransfer);
+            if (!$registryFileCopied) {
+                $result->addError($this->createMessage('Copying local registry file failed'));
+
+                return $result;
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return \cebe\openapi\spec\OpenApi
+     */
+    protected function getDefaultEndpointsSchema(CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer): OpenApi
+    {
+        $filePath = getcwd() . '/config/app/api/openapi/registry_reference_remote.yml';
+
+        if ($createDefaultEndpointsRequestTransfer->getAddLocal()) {
+            $filePath = getcwd() . '/config/app/api/openapi/registry_reference_local.yml';
+        }
+
+        return Reader::readFromYamlFile($filePath, OpenApi::class, false);
+    }
+
+    /**
+     * @param \Transfer\CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer
+     *
+     * @return bool
+     */
+    protected function copyRegistryLocalFile(CreateDefaultEndpointsRequestTransfer $createDefaultEndpointsRequestTransfer): bool
+    {
+        $schemaFileParts = explode('/', $createDefaultEndpointsRequestTransfer->getSchemaFile());
+        array_pop($schemaFileParts);
+        $registryFilePath = implode('/', [...$schemaFileParts, 'registry.yml']);
+
+        return copy(getcwd(). '/config/app/api/openapi/registry.yml', $registryFilePath);
```

Comment:
We will also need to ask the config component about the location of the source file.